lib/reporting/report_metadata_builder.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Reporting
+  class ReportMetadataBuilder
+    attr_reader :report, :current_user
+
+    def initialize(report, current_user = nil)
+      @report = report
+      @current_user = current_user
+    end
+
+    def rows
+      rows = []
+      rows.concat(title_rows)
+      rows.concat(date_range_rows)
+      rows.concat(printed_rows)
+      rows.concat(other_filter_rows)
+      rows << [] # spacer before the sheet
+      rows
+    end
+
+    private
+
+    DATE_FROM_KEYS = %i[completed_at_gt created_at_gt updated_at_gt].freeze
+    DATE_TO_KEYS   = %i[completed_at_lt created_at_lt updated_at_lt].freeze
+
+    def title_rows
+      type = params[:report_type]
+      sub  = params[:report_subtype]
+      return [] unless present?(type)
+
+      title = [type, sub].compact.map { |s| s.to_s.tr('_', ' ').titleize }.join(' – ')
+      [['Report Title', title]]
```

Comment:
"Report Title" should be translatable through I18n.

Reply:
We also have translations for the actual report names. Have a look at the index page.

Reply:
I've translated "Report Title" and implemented the translations for report names, however, there seem to be some inconsistencies for subtypes. For instance the subtype called order_cycle_supplier_totals is not in en.yml and instead is supplier_totals: Order Cycle Supplier Totals.  I compromised by just translating the type and titleizing the subtype.  This can be pretty easily fixed just by comparing these types to the subtype names received, but I don't want to break anything existing.  I think this could be another PR.